### PR TITLE
chore: update steps script to access ci-account config [csre-3144] 

### PR DESCRIPTION
**Pull request by ngocpea**
### Purpose :dart:

This work is part of our initiative to rollout a deploy-templates plugin for our services. 

This PR is linked to https://github.com/cultureamp/continuous-integration/pull/209 and is required to be merged prior to this PR, for the plugin to work

### Context :brain:

These changes allow for the plugin to source config from our ci dependency bucket rather than using a config file from the repo. 

### Testing 🧪 
- [x] These changes have been tested via buildkite agents using the experiemental queue using `cost-insights` as an example 
- [x] Successfully deployed to `cost-insights` https://buildkite.com/culture-amp/cost-insights/builds/170

### CODE REVIEW

File: lib/steps.bash [hunk 1]
```diff
@@ -52,13 +52,21 @@ function write_steps() {
           export "${var_name^^}"="${val}"
         done
 
-        # find env file based on location of template
+        if [[ -n "${BUILDKITE_DEPLOY_TEMPLATE_BUCKET:-}" ]]; then
+          ENV_CONFIG_FILE="${BUILDKITE_DEPLOY_TEMPLATE_BUCKET}/${STEP_ENVIRONMENT}.env"
+          echo "=> Downloading .env file from S3..."
+          echo "loading central config ${ENV_CONFIG_FILE} into environment..."
+          aws s3 cp "${ENV_CONFIG_FILE}" .
+          load_env_file "${STEP_ENVIRONMENT}".env
```
Comment (madobrien) [inline lib/steps.bash:60]: NIT
```suggestion
          load_env_file "${STEP_ENVIRONMENT}.env"
```

File: lib/steps.bash [hunk 2]
```diff
@@ -52,13 +52,21 @@ function write_steps() {
           export "${var_name^^}"="${val}"
         done
 
-        # find env file based on location of template
```
Comment (madobrien) [inline lib/steps.bash:55]: Lets add the comments back in to document both of the env conditionals.

**Review by madobrien — APPROVED** [review]
Nice work!

**Comment by lukefaccin** [comment]
After discussing with Maddy, going to merge this since it's versioned. I won't merge updates to use this version of the plugin until the env file rollout has been fixed.

**Pull request merged**